keanu-project/src/test/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensorTest.java

```diff
@@ -935,6 +936,48 @@ public void canFindElementWiseMinAndMax() {
         assertArrayEquals(new double[]{2, 3, 3, 4}, max.asFlatDoubleArray(), 1e-6);
     }
 
+    @Test
+    public void canArgFindMaxOfVector() {
+        DoubleTensor tensor = DoubleTensor.create(1, 3, 4, 5, 2);
+
+        assertThat(tensor.argMax(), isScalarWithValue(3));
+        assertThat(tensor.argMax(0), tensorEqualTo(IntegerTensor.zeros(5).reshape(5)));
```

Comment:
We shouldn't have a reshape of just `5`, that's not supported by ND4J. It'll need to be a rank 2 vector, `[1,5]` or `[5,1]`